core/src/main/java/com/gentics/mesh/core/data/container/impl/NodeGraphFieldContainerImpl.java

```diff
@@ -724,10 +722,15 @@ public String getSegmentFieldValue() {
 
 		// 3. Try to load the path segment using the binary field since the string field could not be found
 		if (stringField == null) {
-			BinaryGraphField binaryField = getBinary(segmentFieldKey);
-			if (binaryField != null) {
-				return binaryField.getFileName();
+			S3BinaryGraphField s3binaryField = getS3Binary(segmentFieldKey);
+			if (nonNull(s3binaryField) && nonNull(s3binaryField.getS3Binary())) {
+				return s3binaryField.getS3Binary().getFileName();
 			}
+			BinaryGraphField binary = getBinary(segmentFieldKey);
+			if (nonNull(binary) && nonNull(binary.getBinary())) {
```

Comment:
Are you sure regarding the extra check for `binary.getBinary()`?

Reply:
Actually I found this relevant for S3BinaryField but if binary.getBinary() has worked for binary fields, it is better to revert this change. I will revert it and check only if this is not null.